### PR TITLE
iOS Simulator: Attempt 'sideloading' the updated Flutter application if the application runner is already up

### PR DESCRIPTION
**Pull request by chinmaygarde**
- This makes the turnaround times < 2 seconds on the iOS simulator.
- Also bumps the engine to use the new engine required to support the flags.

### CODE REVIEW

**Comment by chinmaygarde** [comment]
screen shot 2016-03-02 at 11 38 06 am

File: packages/flutter_tools/lib/src/ios/simulators.dart [hunk 1]
```diff
@@ -363,6 +352,57 @@ class IOSSimulator extends Device {
...
+      'xcrun',
+      'simctl',
+      'get_app_container',
+      'booted',
+      app.id,
+    ]);
+
+    bool isRunning = exitsHappy([
+      '/usr/bin/killall',
+      'Runner',
```
Comment (eseidelGoogle) [inline packages/flutter_tools/lib/src/ios/simulators.dart]: Do we need to share this constant name somewehre?

Comment (chinmaygarde) [inline packages/flutter_tools/lib/src/ios/simulators.dart]: Yeah, that part if a bit brittle. There is no way to query simctl to check if a process is running by its identifier. I am going to make the name a bit more unique to Flutter. Would need to land a change in the engine first though.

**Comment by chinmaygarde** [comment]
Fixes https://github.com/flutter/flutter/issues/2256 https://github.com/flutter/flutter/issues/873

**Comment by eseidelGoogle** [comment]
lgtm

**Pull request merged**